PORTAL/jobs/_pyperformance.py

```diff
@@ -2357,18 +2709,27 @@ def _read_rows(self):
         except StopIteration:
             raise NotImplementedError(self._indexfile)
         if headerstr != '\t'.join(self.INDEX_FIELDS):
-            raise ValueError(header)
+            raise ValueError(headerstr)
         # Now read the rows.
         return rows
 
-    def _parse_row(self, row):
-        rowstr = row
+    # TYPE_TODO
+    def _parse_row(
+            self,
+            rowstr: str
+    ) -> Tuple[
+        str,
+        PyperfUploadID,
+        Optional[str],
+        Optional[str],
+        Optional[str]
+    ]:
         row = rowstr.split('\t')
         if len(row) != len(self.INDEX_FIELDS):
             raise ValueError(rowstr)
         relfile, uploadid, build, baseline, mean = row
-        uploadid = PyperfUploadID.parse(uploadid)
-        if not uploadid:
+        uploadid_resolved = PyperfUploadID.parse(uploadid)
+        if not uploadid_resolved:
```

Comment:
I thought mypy could do some sort of type tightening in cases like this.

Reply:
`PyperfUploadID.parse` can return `None`, so `mypy` errors if you don't have a `None` check.

Reply:
Yeah, I was thinking `uploadid` was not `str` for some reason.  (Maybe I was starting to drift toward the end of the review. 😄.)  Regardless, in this case it would be better then to rename `uploadid` to `uploadid_str` and change `uploadid_resolved` back to `uploadid`.